lib/KBaseReport/authclient.py

```diff
@@ -40,12 +40,15 @@ def add_valid_token(self, token, user):
             raise ValueError('Must supply token')
         if not user:
             raise ValueError('Must supply user')
-        token = hashlib.sha256(token).hexdigest()
+        token = hashlib.sha256(token.encode('utf-8')).hexdigest()
         with self._lock:
             self._cache[token] = [user, _time.time()]
             if len(self._cache) > self._maxsize:
-                for i, (t, _) in enumerate(sorted(self._cache.items(),
-                                                  key=lambda (_, v): v[1])):
+                sorted_items = sorted(
+                    list(self._cache.items()),
+                    key=(lambda v: v[1][1])
+                )
+                for i, (t, _) in enumerate(sorted_items):
```

Comment:
This kept on throwing compilation errors so I copied the version in `lib/installed_clients/` over. Compilation errors gone!